libraries/AP_Logger/LogFile.cpp

```diff
@@ -1413,14 +1413,10 @@ void AP_Logger::Write_Current_instance(const uint64_t time_us,
             time_us             : time_us,
             voltage             : battery.voltage(battery_instance)
         };
-        for (uint8_t i = 0; i < ARRAY_SIZE(cells.cells); i++) {
+        for (uint8_t i = 0; i < MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN; i++) {
```

Comment:
This is wrong. There are lots of reasons why the internal array may move off of the MAVLink cells to be a larger number, which is why the loop should be using the `ARRAY_SIZE`.

Reply:
Ok I can revert that